packages/editor-ui/src/components/NodeCreator/CategorizedItems.vue

```diff
@@ -0,0 +1,378 @@
+
+			if (e.key === 'ArrowDown') {
+				this.activeIndex++;
+				// Make sure that we stop at the last nodeType
+				this.activeIndex = Math.min(
+					this.activeIndex,
+					activeList.length - 1,
+				);
+			} else if (e.key === 'ArrowUp') {
+				this.activeIndex--;
+				// Make sure that we do not get before the first nodeType
+				this.activeIndex = Math.max(this.activeIndex, 0);
+			} else if (e.key === 'Enter' && activeNodeType) {
+				this.selected(activeNodeType);
+			} else if (e.key === 'ArrowRight' && activeNodeType && activeNodeType.type === 'subcategory') {
+				this.selected(activeNodeType);
+			} else if (e.key === 'ArrowRight' && activeNodeType && activeNodeType.type === 'category' && !(activeNodeType.properties as ICategoryItemProps).expanded) {
+				this.selected(activeNodeType);
+			} else if (e.key === 'ArrowLeft' && activeNodeType && activeNodeType.type === 'category' && (activeNodeType.properties as ICategoryItemProps).expanded) {
+				this.selected(activeNodeType);
+			}
+		},
+		selected(element: INodeCreateElement) {
+			if (element.type === 'node') {
+				this.$emit('nodeTypeSelected', (element.properties as INodeItemProps).nodeType.name);
+			} else if (element.type === 'category') {
+				this.onCategorySelected(element.category);
+			} else if (element.type === 'subcategory') {
+				this.onSubcategorySelected(element);
+			}
+		},
+		onCategorySelected(category: string) {
+			if (this.activeCategory.includes(category)) {
+				this.activeCategory = this.activeCategory.filter(
+					(active: string) => active !== category,
+				);
+			} else {
+				this.activeCategory = [...this.activeCategory, category];
+				this.$telemetry.trackNodesPanel('nodeCreateList.onCategoryExpanded', { category_name: category, workflow_id: this.$store.getters.workflowId });
+			}
+
+			this.activeIndex = this.categorized.findIndex(
+				(el: INodeCreateElement) => el.category === category,
+			);
+		},
+		onSubcategorySelected(selected: INodeCreateElement) {
+			this.$store.commit('ui/setShowNodeCreatorTabs', false);
+			this.activeSubcategoryIndex = 0;
+			this.activeSubcategory = selected;
+			this.$telemetry.trackNodesPanel('nodeCreateList.onSubcategorySelected', { selected, workflow_id: this.$store.getters.workflowId });
+		},
+
+		onSubcategoryClose() {
+			this.activeSubcategory = null;
+			this.activeSubcategoryIndex = 0;
+			this.nodeFilter = '';
+			this.$store.commit('ui/setShowNodeCreatorTabs', true);
+			this.$emit('subcategoryClose');
+		},
+
+		onClickInside() {
+			this.searchEventBus.$emit('focus');
+		},
+	},
+	async mounted() {
+		this.$nextTick(() => {
+			// initial opening effect
+			this.activeCategory = [CORE_NODES_CATEGORY];
+		});
+		this.$externalHooks().run('nodeCreateList.mounted');
+	},
+	async destroyed() {
+		this.$externalHooks().run('nodeCreateList.destroyed');
+		this.$telemetry.trackNodesPanel('nodeCreateList.destroyed', { workflow_id: this.$store.getters.workflowId });
```

Comment:
these hooks and telemetry logic is now duplicated across the components, which could create issues for telemetry and hooks.. just delete from here.. no need for new hooks